bolt-modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -119,8 +119,15 @@ def run_task_raw(task_name, targets, description = nil, task_args = nil, &block)
       task = task_signature.task
     end
 
-    unless Puppet::Pops::Types::TypeFactory.data.instance?(use_args)
-      raise with_stack(:TYPE_NOT_DATA, 'Task parameters is not of type Data')
+    unless Puppet::Pops::Types::TypeFactory.rich_data.instance?(use_args)
```

Comment:
I'm not familiar enough with the semantics of RichData yet. Tasks require input that is JSON-serializable, which is why we initially restricted it to Data. If we're accepting more, that seems to imply some sort of transform that needs to be applied when passing the arguments to the task. Is that transform always clear?

Reply:
I'm concerned that we aren't careful enough with how we treat data downstream of the run functions yet and it may be best if the plan author has to explicitly unwrap sensitive data until we are.

Reply:
How do you then invoke a task whose parameter is of type `Sensitive[String]` from a plan?

Tasks will need to then modify their parameter types to be `String`, otherwise type checks will fail earlier on in the `run_task` function.

Reply:
We need to consider carefully how the two sensitives interact.

I think this is roughly what they mean.
1. When a parameter is marked sensitive Bolt should not log it anywhere inside "run task".
2. When a value in a plan is marked sensitive Bolt should not log it anywhere during the plan execution

How do they interact though?
It seems obvious that a sensitive value should be unwrapped automatically when given to a sensitive param. The task has opted into treating the parameter sensitively.
Should a sensitive value be unwrapped automatically into a non-sensitive param? The task may leak it.